pkg/apis/core/validation/validation.go

```diff
@@ -4127,13 +4127,18 @@ func ValidatePodSpec(spec *core.PodSpec, podMeta *metav1.ObjectMeta, fldPath *fi
 		gracePeriod = *spec.TerminationGracePeriodSeconds
 	}
 
+	var inUserNS bool
+	if spec.SecurityContext != nil && spec.SecurityContext.HostUsers != nil && !*spec.SecurityContext.HostUsers {
```

Comment:
Should we also guard this with a feature gate?

Reply:
Did a quick search for hostUsers and we don't guard this validation with a feature gate so I think this is fine actually.

Reply:
we do already do some validation on the proc mount field in this file, and there's no mention of the feature, which implies to me the field is already [filtered out](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/pod/util.go#L775) if ProcMount isn't an available option (or it doesn't matter if it's done before, as happening after works too)

Reply:
Sounds good to me.